### PR TITLE
Implement flush_all for the ASCII protocol

### PR DESCRIPTION
**Pull request by oosidat**
### TL;DR - What are you trying to accomplish?

Adds the `flush_all` memcached operation to async-memcached. Supports the simple case only, without the optional delay.

[Memcached Protocol for `flush_all`](https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L1796-L1818)

### Details - How are you making this change?  What are the effects of this change?
- Add the `flush_all` method that sends the `flush_all` Memcached command
- Uses the `serial_test` crate to mark the `flush_all` test as serial (and others as parallel) to prevent the `flush_all` test from interfering with other tests.

### CODE REVIEW

**Review by drinkbeer — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -439,6 +439,28 @@ impl Client {
...
+    /// This operation invalidates all existing items immediately. Any items with an update time
+    /// older than the time of the flush_all operation will be ignored for retrieval purposes.
+    /// This operation does not free up memory taken up by the existing items.
+
+    pub async fn flush_all(&mut self) -> Result<(), Error> {
+        self.conn.write_all(b"flush_all\r\n").await?;
+        self.conn.flush().await?;
+
+        let mut response = String::new();
+        self.conn.read_line(&mut response).await?;
```
Comment (drinkbeer) [inline src/lib.rs:454]: Could we reuse the `self.buf` rather than create a new string as buffer?

```
self.conn.read_buf(&mut self.buf).await?;
```

File: src/lib.rs [hunk 2]
```diff
@@ -491,6 +514,7 @@ mod tests {
 
     #[ignore = "Relies on a running memcached server"]
     #[tokio::test]
+    #[parallel]
```
[the remark this answers is not in the document]
Comment (mrattle) [reply src/lib.rs:517]: I think the annotation is required on all tests or they won't differentiate properly, unfortunately.  Everything already has to be annotated with `ignore` and `tokio::test` for now, anyway.

File: src/lib.rs [hunk 3]
```diff
@@ -439,6 +439,28 @@ impl Client {
...
+        let mut response = String::new();
+        self.conn.read_line(&mut response).await?;
+        // check if response is ok
+        if response.trim() == "OK" {
+            Ok(())
+        } else {
+            Err(Error::from(Status::Error(ErrorKind::Protocol(Some(
+                format!("Invalid response for `flush_all` command: `{response}`"),
+            )))))
+        }
```
Comment (mrattle) [inline src/lib.rs:462]: Played around with the existing parser to see if we could leverage that but it looks like it's not built to handle this type of response at the moment.  This works fine for now but it might be better overall to lean in to the parser architecture (potentially in a later PR)?

**Review by mrattle — APPROVED** [review]
LGTM, just left a non-blocking thought in a comment.

File: src/lib.rs (hunk 3 above)
Comment (oosidat) [reply src/lib.rs:462]: I mostly went with this approach since I saw it on the `version` call. I had a full changeset with `Status::Ok` and all, but it felt a little overkill.

File: src/lib.rs (hunk 1 above)
Comment (oosidat) [reply src/lib.rs:454]: I think you'll need to create a new string anyways to check if "OK"?

**Review by nickamorim — APPROVED** [review]
